pkg/collector/collector_test.go

```diff
@@ -0,0 +1,62 @@
+package collector
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/archway-network/relayer_exporter/pkg/config"
+)
+
+func TestDiscordIDs(t *testing.T) {
+	testCases := []struct {
+		name     string
+		ops      []config.Operator
+		expected string
+	}{
+		{
+			name: "All Valid IDs",
+			ops: []config.Operator{
+				{
+					Discord: config.Discord{ID: "123456"},
+				},
+				{
+					Discord: config.Discord{ID: "12312387"},
+				},
+			},
+			expected: "123456,12312387",
+		},
+		{
+			name: "Some Invalid IDs",
+			ops: []config.Operator{
+				{
+					Discord: config.Discord{ID: "123456"},
+				},
+				{
+					Discord: config.Discord{ID: "ABCDEF"},
+				},
+				{
+					Discord: config.Discord{ID: "789012"},
+				},
+			},
+			expected: "123456,789012",
+		},
+		{
+			name: "No Valid IDs",
+			ops: []config.Operator{
+				{Discord: config.Discord{ID: "ABCDEF"}},
+				{Discord: config.Discord{ID: "GHIJKL"}},
+			},
+			expected: "",
+		},
+		{
+			name:     "Empty Input",
+			ops:      []config.Operator{{}, {}, {}},
+			expected: "",
+		},
+	}
+	for _, tc := range testCases {
+		res := getDiscordIDs(tc.ops)
```

Comment:
I recommend to wrap it in `t.Run` like
```
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			res := getDiscordIDs(tc.ops)
			assert.Equal(t, tc.expected, res)
		})
	}
```
This way it is easy to spot which test case failed and a test case name is being used. To show it I've made tests failing like
```
=== RUN   TestDiscordIDs
    collector_test.go:60:
        	Error Trace:	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector_test.go:60
        	Error:      	Not equal:
        	            	expected: "123456,12312387"
        	            	actual  : "12312387"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-123456,12312387
        	            	+12312387
        	Test:       	TestDiscordIDs
    collector_test.go:60:
        	Error Trace:	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector_test.go:60
        	Error:      	Not equal:
        	            	expected: ","
        	            	actual  : ""

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-,
        	            	+
        	Test:       	TestDiscordIDs
--- FAIL: TestDiscordIDs (0.00s)
```
vs
```
=== RUN   TestDiscordIDs
=== RUN   TestDiscordIDs/All_Valid_IDs
    collector_test.go:61:
        	Error Trace:	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector_test.go:61
        	Error:      	Not equal:
        	            	expected: "123456,12312387"
        	            	actual  : "12312387"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-123456,12312387
        	            	+12312387
        	Test:       	TestDiscordIDs/All_Valid_IDs
=== RUN   TestDiscordIDs/Some_Invalid_IDs
=== RUN   TestDiscordIDs/No_Valid_IDs
=== RUN   TestDiscordIDs/Empty_Input
    collector_test.go:61:
        	Error Trace:	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector_test.go:61
        	Error:      	Not equal:
        	            	expected: ","
        	            	actual  : ""

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-,
        	            	+
        	Test:       	TestDiscordIDs/Empty_Input
--- FAIL: TestDiscordIDs (0.00s)
```
I think in the second report it is easier to understand which test case failed.

Reply:
Yes, makes sense!